packages/firebase/index.d.ts

```diff
@@ -7080,13 +7080,73 @@ declare namespace firebase.firestore {
    *
    *   <ul>
    *     <li><code>debug</code> for the most verbose logging level, primarily for
-   *     dubugging.</li>
+   *     debugging.</li>
    *     <li><code>error</code> to log errors only.</li>
    *     <li><code>silent</code> to turn off logging.</li>
    *   </ul>
    */
   export function setLogLevel(logLevel: LogLevel): void;
 
+  /**
+   * Converter used by `withConverter()` to transform user objects into
+   * Firestore data.
+   *
+   * Using the converter allows you to specify generic type arguments when
+   * storing and retrieving objects from Firestore.
+   *
+   * @example
+   * ```typescript
+   * class Post {
+   *   constructor(readonly title: string, readonly author: string) {}
+   *
+   *   toString(): string {
+   *     return this.title + ', by ' + this.author;
+   *   }
+   * }
+   *
+   * const postConverter = {
+   *   toFirestore(post: Post): firebase.firestore.DocumentData {
+   *     return {title: post.title, author: post.author};
+   *   },
+   *   fromFirestore(
+   *     snapshot: firebase.firestore.QueryDocumentSnapshot,
+   *     options: firebase.firestore.SnapshotOptions
+   *   ): Post {
+   *     const data = snapshot.data(options)!;
+   *     return new Post(data.title, data.author);
+   *   }
+   * };
+   *
+   * const postSnap = await firebase.firestore()
+   *   .collection('posts')
+   *   .withConverter(postConverter)
+   *   .doc().get();
+   * const post = postSnap.data();
+   * if (post !== undefined) {
```

Comment:
I think specifying the custom types in the code snippet provides an exact example of what custom types are and what they enable the user to do that they otherwise could not.